packages/plugins/type-designer/src/components/element-type.svelte

```diff
@@ -0,0 +1,159 @@
+import ContentCard from './content-card.svelte'
+// STORES
+import { drawerStore } from '@oscd-plugins/ui'
+import { dataTypeTemplatesStore } from '@/stores/data-type-templates.store'
+// TYPES
+import type { DataTypeTemplates } from '@oscd-plugins/core'
+
+//====== INITIALIZATION ======//
+
+// props
+export let typeElement: DataTypeTemplates.TypeElement
+export let columnKey: keyof typeof SCD_ELEMENTS
+
+// store
+const { drawer } = drawerStore
+const { dataTypeTemplatesSubElements } = dataTypeTemplatesStore
+
+// local
+let isCurrentPanelOpen = false
+
+//====== FUNCTIONS ======//
+
+function handleCardClick(
+	currentTypeOrRefElement:
+		| DataTypeTemplates.TypeElement
+		| DataTypeTemplates.TypeRefElement
+) {
+	const { currentColumnKey, currentTypeElement } = getCurrentTypeElement(
+		currentTypeOrRefElement
+	)
+	console.log(currentColumnKey, currentTypeElement)
+	const payload = {
+		title: `Edit ${SCD_ELEMENTS[currentColumnKey].element.name} type`,
+		description: `#${currentTypeElement.id}`,
+		component: DrawerContent,
+		componentProps: {
+			currentTypeElement,
+			currentColumnKey
+		}
+	}
+
+	if (
+		$drawer.componentProps?.currentTypeElement.id &&
+		$drawer.componentProps?.currentTypeElement.id !== currentTypeElement.id
+	)
+		return drawerStore.handleOpenDrawer(payload)
+
+	return drawerStore.handleSwitchDrawer(payload)
+}
+
+//====== FUNCTIONS ======//
+
+function isTypeElement(
+	typeOrRef: DataTypeTemplates.TypeElement | DataTypeTemplates.TypeRefElement
+): typeOrRef is DataTypeTemplates.TypeElement {
+	return (typeOrRef as DataTypeTemplates.TypeElement).name !== undefined
+}
+
+function getCurrentTypeElement(
+	currentTypeOrRefElement:
+		| DataTypeTemplates.TypeElement
+		| DataTypeTemplates.TypeRefElement
+): {
+	currentColumnKey: keyof typeof SCD_ELEMENTS
+	currentTypeElement: DataTypeTemplates.TypeElement
+} {
+	if (isTypeElement(currentTypeOrRefElement))
+		return {
+			currentColumnKey: columnKey,
+			currentTypeElement: currentTypeOrRefElement
+		}
+
+	const currentTypeRefColumn = SCD_ELEMENTS[columnKey].typeRef.from
+	if (!currentTypeRefColumn) throw new Error('TypeRef column not found')
+
+	const typeElementFromRef = $dataTypeTemplatesSubElements[
+		`${currentTypeRefColumn}Types`
+	].find((subElement) => subElement.id === currentTypeOrRefElement.type)
+
+	if (!typeElementFromRef) throw new Error('Type element not found')
+	return {
+		currentColumnKey: currentTypeRefColumn,
+		currentTypeElement: typeElementFromRef
+	}
+}
+
+function getTypeOrRefName(
+	currentTypeOrRefElement:
+		| DataTypeTemplates.TypeElement
+		| DataTypeTemplates.TypeRefElement
+) {
+	const { currentTypeElement } = getCurrentTypeElement(
+		currentTypeOrRefElement
+	)
+
+	return currentTypeElement.name || currentTypeElement.id
+}
+</script>
+
+<section id="type-designer-card">
+	<Accordion  class="accordion">
+		{#if typeElement.typeRefs?.length}
+			<Panel bind:open={isCurrentPanelOpen}>
+				<Header>
```

Comment:
Header and PanelContent are two components that are responsible for an incompatibility bug below

<img width="3200" alt="image" src="https://github.com/user-attachments/assets/5ece9efa-fea4-452d-aa4f-108fcb8c30ed">

Reply:
As this is linked to another future feature, and as the development is on hold for this plugin : I won't inspect this bug for now, which is out of scope.